### ISSUE TITLE
upgrade dialog to select recipe  appears twice

### ISSUE THREAD

**Issue by martinlippert** [spring-projects/sts4#1245]
Spring Tools 4 on Eclipse `4.22.1.RELEASE`, empty workspace. Created `Rest Service` guide project.

When I right-click on the project or pom file and execute the `Spring -> Upgrade Spring Boot Version...´, the upgrade dialog where I can select the recipe appears nicely.

When I execute the quick fix to upgrade the boot version first (from the problems view, for example), and then click on the action in the context menu to upgrade spring boot versions, the dialog window appears twice. One of the windows appears to be working (gets the list of recipes in an async way and displays the list), the other one is stuck in the loading phase (at least from the UI perspective).

**Comment by BoykoAlex** [comment]
This happens because the command for listing recipes is attempted to be executed on the XML LS as well or on any LS started from Eclipse :-\
Turns out that `computeFirst` on language executors really takes the result from the first LS (LS Wrapper) from execution of the request/command on the LS but doesn't skip executing the same on the rest of the LSs. `preferServerDefinition` only moves the server you'd like to execute request/command on to the top of the list. Therefore to fix this I check if the command is supported by the server as well via server capabilities filter - luckily this command is registered at the server initialization stage.
I have disabled XML LS hence didn't pay enough attention.
